code/modules/mob/living/brain/brain_item.dm

```diff
@@ -60,6 +60,8 @@
 	name = "[L.name]'s brain"
 	if(brainmob || decoy_override)
 		return
+	if(L.mind.current != L)
```

Comment:
Prevents the creation of a brainmob when the body's mind has already been transferred to a new mob.